mitodl/ocw-studio
# Decoupled 'websites' app from import code

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #47

#### What's this PR do?
- Moves OCW import code to its own Django app
- Renames `Website.url_path` to `name`
- Adds a data migration to create a `WebsiteStarter` record for our OCW starter repo
- Removes `Website.type`, opting instead to get the website's de facto "type" from the associated `WebsiteStarter.slug`
- Adds the `Website.source` field to identify which sites were imported and which ones were authored in the app
- Adds a bunch of data migrations to ensure that imported Websites are handled correctly and can still be imported

#### How should this be manually tested?
- Run the OCW import command to test that it still works (ex: `backpopulate_ocw_courses -b <bucket> --filter "esd-7`). It would be best to delete a `Website` and re-import it.
- Check through imported `Website` objects in Django admin to make sure the data looks correct
- Make sure that requests to `/api/websites?type=course` are unchanged
